File: imdb.R
rm(list=ls())
getwd()
setwd('~/Desktop/Edwisor/Predictive analysis using R and Python/Basic/data01s2l1/')
df=read.csv('IMDB_data.csv',header = T,fileEncoding="latin1",skip=0)
df=df[-c(2),]
rownames(df)=c(1:3389)

b=table(df$Genre)
b=data.frame(b)

df$imdbRating=as.numeric(df$imdbRating)
df$imdbVotes=as.numeric(df$imdbVotes)
df$Year=as.numeric(df$Year)
df$imdbID=as.numeric(df$imdbID)

df=df[order(df$Genre),]

df$newVar=(df$imdbRating - df$imdbVotes)^2

File: imdb.py
import pandas as pd 

df=pd.read_csv("IMDB_data.csv",encoding="ISO-8859-1",skiprows=[2],header=[0])
a=df['Genre'].value_counts()
b=a.keys()
c=[]
for i in b:
	c=c+[a[i]]

df1=pd.DataFrame({'Genre':b,'Freq':c})

for i in range(0,len(df['imdbVotes'])):
	try:
		df['imdbVotes'][i]=int(df['imdbVotes'][i])
	except AttributeError:
		df['imdbVotes'][i]=0
	except ValueError:
		df['imdbVotes'][i]=0	

for i in range(0,len(df['imdbRating'])):
	try:
		df['imdbRating'][i]=int(df['imdbRating'][i])
	except AttributeError:
		df['imdbRating'][i]=0
	except ValueError:
		df['imdbRating'][i]=0	

for i in range(0,len(df['Year'])):
	try:
		df['Year'][i]=int(df['Year'][i])
	except AttributeError:
		df['Year'][i]=0
	except ValueError:
		df['Year'][i]=0

print(df.sort_values(['Genre'], ascending=[True]))
df['newVar']=(df['imdbRating']-df['imdbVotes'])*(df['imdbRating']-df['imdbVotes'])
print(df['newVar'])
